src/edu/gatech/cs2340/trydent/TrydentEngine.java

```diff
@@ -0,0 +1,145 @@
+package edu.gatech.cs2340.trydent;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Main class that kicks off the TrydentEngine.
+ * 
+ * As a developer using the TrydentEngine, all you have to do is call
+ * TrydentEngine.start() to get started.
```

Comment:
So a benefit of TrydentEngine being a singleton is that anytime a user interacts with it, getInstance() is called internally. Which means we could easily automatically call start() there if we wanted to invoke it lazily in the future. So perhaps we should leave it as-is for now, and also leave TrydentEngine as a singleton so that we can easily add that later.